test/input-rttc.test.js

```diff
@@ -160,6 +206,59 @@ describe('argin validation & "light coercion"  (for inputs)', function (){
     });
 
 
+    it('should error if an invalid type is given for an input', function() {
+      var machine = {
+        inputs: {
+          foo: {
+            type: 'foobar'
+          }
+        },
+        exits: {
+          success: {
+            example: 'hello'
+          },
+          error: {
+            example: 'world'
+          }
+        },
+        fn: function (inputs, exits, deps) {
+          exits(null, 'foo');
+        }
+      };
+
+      assert.throws(function() {
+        M.build(machine);
+      }, Error);
+    });
+
+
+    it('should error if an input has an incompatible type / example combo', function() {
+      var machine = {
+        inputs: {
+          foo: {
+            type: 'number',
+            example: 'abc'
+          }
+        },
+        exits: {
+          success: {
+            example: 'hello'
+          },
+          error: {
+            example: 'world'
+          }
+        },
+        fn: function (inputs, exits, deps) {
+          exits(null, 'foo');
+        }
+      };
+
+      assert.throws(function() {
```

Comment:
I'd avoid using assert.throws() -- that thing has screwed us so many times.  How about:

```javascript
try {
  //...
} catch (e) {
  switch (e.code) {
    case 'MACHINE_INPUT_INVALID': break; // ok
    default: throw new Error('Expected MACHINE_INPUT_INVALID error, but instead got `code: '+e.code+'`.  Details: '+e.stack);
  }
}